### PR TITLE
Feature/outlined input

### LINKED ISSUE #1752: Outlined Inputs
**Enhancement**

Add support for outlined inputs as per the material.io guidelines https://material.io/components/text-fields#input-types.

This should use a property `kind` to determine the type of input and supported by all the different types of input such as email-input, password-input, select etc

### PR DESCRIPTION
**Pull request by KaneFreeman**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [ ] For new widgets, an entry has been added to the `.dojorc`
* [ ] For new widgets, `theme.variant()` is added to the root domnode
* [ ] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [ ] WidgetProperties are exported

**Description:**
Add outlined option "kind" to `text-input` and `text-area` widgets

**Text Input**
text-input

**Text Area**
text-area

Resolves #1752

### CODE REVIEW

File: src/text-area/index.tsx [hunk 1]
```diff
@@ -196,6 +201,60 @@ export const TextArea = factory(function TextArea({
 	const [labelChild] = children();
 	const label = isRenderResult(labelChild) ? labelChild : labelChild.label;
 
+	let {
+		size: { width: labelWidth }
+	} = dimensions.get('label');
+
+	// When notched the label is transformed in the CSS to 75% size, then we need to add 10px of padding to get the desired look
+	labelWidth = labelWidth * 0.75 + 10;
```
Comment (tomdye) [inline src/text-area/index.tsx:209]: is this the same with dojo theme too?

**Review by tomdye — APPROVED** [review]

**Comment by agubler** [comment]
@KaneFreeman Looks solid for these inputs, but we do need to add the same to Select and ChipTypeahead as well as they are inputs too.

The styling for the existing chip-typeahead seems to have been affected by the changes too https://widget-test-do-git-fork-kanefreeman-feature-outlined-inp-f7d95a.vercel.app/#widget/chip-typeahead/overview

File: src/text-area/index.tsx (hunk 1 above)
Comment (KaneFreeman) [reply src/text-area/index.tsx:209]: Sadly no. Is there a way to detect the theme name? I did not see a way. I got around this by using CSS for override the width of the label one the dojo theme with an `important!` tag.

Comment (tomdye) [reply src/text-area/index.tsx:209]: Can't you just override the transform in the material theme CSS?

Comment (KaneFreeman) [reply src/text-area/index.tsx:209]: Are you suggesting removing the 75% text size transform from the material theme?

Comment (tomdye) [reply src/text-area/index.tsx:209]: not necessarily, I'm just saying that you should be doing this adjustment, whatever it needs to be, in the material theme css and not in this widget js.

**Pull request closed without merging**